core/src/main/scala/com/rallyhealth/vapors/factfilter/dsl/Dsl.scala

```diff
@@ -17,20 +17,8 @@ private[dsl] class Dsl extends TypedFactOps {
     ExpAlg.Exists[T, V, Boolean](ev, cond, True, False)
   }
 
-  def lessThan[T : Ordering](upperBound: T): CondExp[T] = liftCondExp {
-    ExpAlg.Within[T, Boolean](Window.lessThan(upperBound), True, False)
-  }
-
-  def lessThanOrEqual[T : Ordering](upperBound: T): CondExp[T] = liftCondExp {
-    ExpAlg.Within[T, Boolean](Window.lessThanOrEqual(upperBound), True, False)
-  }
-
-  def greaterThan[T : Ordering](lowerBound: T): CondExp[T] = liftCondExp {
-    ExpAlg.Within[T, Boolean](Window.greaterThan(lowerBound), True, False)
-  }
-
-  def greaterThanOrEqual[T : Ordering](lowerBound: T): CondExp[T] = liftCondExp {
-    ExpAlg.Within[T, Boolean](Window.greaterThanOrEqual(lowerBound), True, False)
+  def within[T](window: Window[T]): CondExp[T] = liftCondExp {
+    ExpAlg.Within[T, Boolean](window, True, False)
```

Comment:
OK, we have some serious case of boolean blindness here and it is too easy to make mistakes.

Maybe add:
* `excludeLowerBound`
* `includeLowerBound`
* `excludeUpperBound`
* `includeUpperBound`

Reply:
These cases are all covered by the WindowSpec.

Reply:
I suppose what I'm getting at is that by looking at this expression: 
`expAlg.Within[T, Boolean](window, True, False)` I don't know what the `True` and `False` represent because this single bit of information has been stripped from its context.

I believe that I reviewed the PR that included the WindowSpec, and yes, it does test all the cases, but it wasn't without error initially (as I recall).

What I'm saying is have types so that instead of saying true/false,  you can put something with more context.

Reply:
Ok, I see how this is working now....  never mind :)